### PR TITLE
[v1] Add AST factory methods; some fixes to AST classes

### PR DESCRIPTION
**Pull request by alancai98**
## Relevant Issues
- https://github.com/partiql/partiql-lang-kotlin/issues/1610

## Description
- Adds factory methods for every hand-written AST class
- Fixes some AST classes/functions
  - Add constructors for some classes
  - Fix naming of `visitSelectItemAll` -> `visitSelectItemStar`
  - Change `@NotNull Long` -> `long`

## Other Information
- Updated Unreleased Section in CHANGELOG: **[NO]**
  - No, targeting v1 branch

- Any backward-incompatible changes? **[YES]**
  - Yes but on v1 branch

- Any new external dependencies? **[NO]**

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **[YES]**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-ast/src/main/java/org/partiql/ast/v1/graph/GraphSelector.java [hunk 1]
```diff
@@ -36,6 +36,8 @@ public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
      */
     @Builder(builderClassName = "Builder")
     public static class AnyShortest extends GraphSelector {
+        public AnyShortest() {}
```
Comment (alancai98) [inline partiql-ast/src/main/java/org/partiql/ast/v1/graph/GraphSelector.java:39]: (self-review) some missing constructors in the graph/ directory.

File: partiql-ast/src/main/java/org/partiql/ast/v1/AstVisitor.java [hunk 2]
```diff
@@ -149,7 +149,7 @@ public interface AstVisitor<R, C> {
 
     R visitSelectItem(SelectItem node, C ctx);
 
-    R visitSelectItemAll(SelectItem.Star node, C ctx);
```
Comment (alancai98) [inline partiql-ast/src/main/java/org/partiql/ast/v1/AstVisitor.java:152]: (self-review) I forgot to change naming of this visitor function after the class was renamed.

File: partiql-ast/src/main/java/org/partiql/ast/v1/graph/GraphQuantifier.java [hunk 3]
```diff
@@ -14,13 +14,12 @@
  */
 @Builder(builderClassName = "Builder")
 public class GraphQuantifier extends AstNode {
-    @NotNull
-    public final Long lower;
```
Comment (alancai98) [inline partiql-ast/src/main/java/org/partiql/ast/v1/graph/GraphQuantifier.java:18]: (self-review) type could just be primitive `long`.

File: partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+
+    @NotNull
+    public static ExprMatch exprMatch(@NotNull Expr expr, @NotNull GraphMatch pattern) {
+        return new ExprMatch(expr, pattern);
+    }
+
+    @NotNull
+    public static ExprNot exprNot(@NotNull Expr value) {
+        return new ExprNot(value);
+    }
```
Comment (alancai98) [inline partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java:137]: (self-review) for PartiQL expression `NOT false`,

example usage from Kotlin
```Kotlin
exprNot(exprLit(boolValue(false)))
```

example usage from Java
```Java
exprNot(exprLit(boolValue(false)));
```

File: partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+import java.util.List;
+import java.util.Map;
+
+// TODO docs for all factory methods
+public class Ast {
+    // Expr
+    @NotNull
+    public static ExprAnd exprAnd(Expr lhs, Expr rhs) {
+        return new ExprAnd(lhs, rhs);
+    }
```
Comment (alancai98) [inline partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java:61]: One potentially annoying thing I've noticed about these factory methods being written in Java is that they cannot be called from Kotlin with named parameters.

E.g. below results in an error "Named arguments are not allowed for non-Kotlin functions"
```
identifier(
    symbol = ctx.IDENTIFIER_QUOTED().getStringValue(),
    isDelimited = true
)
```

Wondering if we should actually have these factory methods written in Kotlin instead?

**Review by RCHowell — APPROVED** [review]

File: partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java (hunk 5 above)
Comment (alancai98) [reply partiql-ast/src/main/java/org/partiql/ast/v1/Ast.java:61]: I'll bring up next week w/ the team if these should be written in Kotlin.

**Pull request merged**